src/app/core/services/issue.service.ts

```diff
@@ -329,22 +331,71 @@ export class IssueService {
     // const issuesAPICallsByFilter = filters.map(filter => this.githubService.fetchIssuesGraphql(filter));
     return forkJoin(issuesAPICallsByFilter).pipe(
       map((issuesByFilter: [][]) => {
+        const fetchedIssueIds: Array<Number> = [];
+
         for (const issues of issuesByFilter) {
           for (const issue of issues) {
+            fetchedIssueIds.push(this.createIssueModel(issue).id);
             this.createAndSaveIssueModel(issue);
           }
         }
+
+        return fetchedIssueIds;
+      }),
+      reduce((acc, ids) => acc.concat(ids), []),
+      map((ids: Array<Number>) => {
+        const outdatedIssueIds: Array<Number> = this.getOutdatedIssueIds(ids);
+        this.deleteIssuesFromLocalStore(outdatedIssueIds);
+
         return Object.values(this.issues);
       })
```

Comment:
Should we remove this line 345? I feel the `reduce` is not necessary here, tested by removing it and works fine.

Also, should we combine the `map` from 346-351 here into one with the previous `map` for less complexity?
```javascript
map((issuesByFilter: [][]) => {
  const fetchedIssueIds: Array<Number> = [];

  for (const issues of issuesByFilter) {
    for (const issue of issues) {
      fetchedIssueIds.push(this.createIssueModel(issue).id);
      this.createAndSaveIssueModel(issue);
    }
  }

  const outdatedIssueIds: Array<Number> = this.getOutdatedIssueIds(fetchedIssueIds);
  this.deleteIssuesFromLocalStore(outdatedIssueIds);

  return Object.values(this.issues);
})
```
I tested and this should work too!

Reply:
Good point. I think i put the reduce there initially because I was trying another way to fix it. Will remove it